scvi/data/anndata/fields/_obsm_field.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from typing import List, Optional
+
+import pandas as pd
+from anndata import AnnData
+
+from scvi.data.anndata import _constants
+
+from ._base_field import BaseAnnDataField
+
+logger = logging.getLogger(__name__)
+
+
+class BaseObsmField(BaseAnnDataField):
+    """An abstract AnnDataField for .obsm attributes in the AnnData data structure."""
+
+    _attr_name = _constants._ADATA_ATTRS.OBSM
+
+    def __init__(
+        self,
+        registry_key: str,
+    ) -> None:
+        super().__init__()
+        self._registry_key = registry_key
+
+    @property
+    def registry_key(self):
+        return self._registry_key
+
+    @property
+    def attr_name(self):
+        return self._attr_name
+
+
+class JointObsField(BaseObsmField):
+    def __init__(self, registry_key: str, obs_keys: Optional[List[str]]) -> None:
+        super().__init__(registry_key)
+        self._attr_key = f"_scvi_{registry_key}"
+        self._columns_key = f"{self.registry_key}_keys"
+        self._obs_keys = obs_keys if obs_keys is not None else []
+        self._is_empty = len(self.obs_keys) == 0
+
+    def _combine_obs_fields(self, adata: AnnData) -> None:
+        series = []
+        for key in self.obs_keys:
+            s = adata.obs[key]
+            series.append(s)
+        adata.obsm[self.attr_key] = pd.concat(series, axis=1)
```

Comment:
this is probably inefficient, should be able to subset obs all at once

Reply:
what do you mean by this? I made it into a generator expression just to make it concise, but don't think its any faster

Reply:
like if you do `adata.obsm[self.attr_key] = adata.obs[[self._obs_keys]].copy()`?